src/test/java/cz/muni/fi/dao/DriverDaoTest.java

```diff
@@ -0,0 +1,118 @@
+import cz.fi.muni.pa165.PersistenceSampleApplicationContext;
+import cz.muni.fi.entities.Driver;
+import cz.muni.fi.dao.DriverDao;
+import cz.muni.fi.dao.DriverDaoImpl;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.Test;
+import static org.assertj.core.api.Assertions.*;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+/**
+ * @author Lucie Kureckova, 445264
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+public class DriverDaoTest extends AbstractTestNGSpringContextTests {
+    
+    @PersistenceContext
+    private EntityManager em;
+    
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+    
+    @Autowired
+    private DriverDao driverManager = new DriverDaoImpl();
+    
+    private Driver mainDriver1 = new Driver();
+    private Driver testDriver1 = new Driver();
+    private Driver testDriver2 = new Driver();
+    private Driver driver1 = new Driver();
+    private Driver driver2 = new Driver();
```

Comment:
Are driver1, driver2, and driver3 needed to be class properties instead of local properties in methods?